lib/diagnostic.js

```diff
@@ -38,19 +50,47 @@ let owner = null;
 // Methods: vocalize
 
 // Replace null with your answer
-let pet = null;
+let pet = {
+  name: 'Briscoe',
+  species: 'cat',
+  breed: 'ragdoll',
+  vocalization: 'meow',
+  vocalize: function () {
+    return this.vocalization;
+  },
+};
 
 // 4. Create a constructor function for pet objects, as described above. The
 // constructor function should be responsible for setting the values of `name`,
 // `species`, `breed`, and `vocalization`.
 
-const Pet = function () {
-  /* Your answer here */
+const Pet = function Pet(name, species, breed) {
+  this.name = name;
+  this.species = species;
+  this.breed = breed;
+  this.vocalization = [];
+};
+
+const Vocalization = function Vocalization(date, soundsLike, mood) {
+  this.date = date;
+  this.soundsLike = soundsLike;
+  this.mood = mood;
 };
 
 // 5. Attach a `vocalize` method to the Pet's prototype.
 
-/* your answer here */
```

Comment:
This is overly complicated and unnecessary.  We just want ```vocalize()``` to return the object's ```vocalization``` value.
```javascript
Pet.prototype.vocalize = function() {
  return this.vocalization;
}
```